aiidalab_eln/eln/cheminfopy.py

```diff
@@ -0,0 +1,59 @@
+import traitlets
+import ipywidgets as ipw
+from . import ElnConnector
+from cheminfopy import User
+from IPython.display import clear_output
+
+class CheminfoElnConnector(ElnConnector):
+    access_token = traitlets.Unicode()
+    
+    def __init__(self, **kwargs):
+        eln_instance_widget = ipw.Text(
+            description="ELN address:",
+            value="",
+            style={"description_width": "initial"},
+        )
+        traitlets.link((self, "eln_instance"), (eln_instance_widget, "value"))
+        
+        
+        token_widget = ipw.Text(
+            description="Token:",
+            value="",
+            placeholder="Press the \"Request token\" button below",
+            style={"description_width": "initial"},
+        )
+        traitlets.link((self, "access_token"), (token_widget, "value"))
+        self.button_clicked = True
+        request_token_button = ipw.Button(description="Request token")
+        request_token_button.on_click(self.request_token)
+        self.output = ipw.Output() 
+        super().__init__(children=[eln_instance_widget, token_widget, request_token_button, self.output], **kwargs)
+        
+    def connect(self):
+        """Connect to the cheminfo ELN."""
+        self.session = User(instance=self.eln_instance, token=self.access_token)
+    
+    def get_config(self):
+        return {
+            "eln_instance": self.eln_instance,
+            "eln_type": self.eln_type,
+            "access_token": self.access_token
+        }
+    
+    def request_token(self, new=None):
```

Comment:
when would `new` be used?

Reply:
actually, it won't be used. Thanks for pointing it out!